packages/ruby-nokogiri/package.py

```diff
@@ -14,8 +14,15 @@ class RubyNokogiri(RubyPackage):
     homepage = "https://nokogiri.org"
     url      = "https://rubygems.org/downloads/nokogiri-1.12.5.gem"
 
+    version('1.16.7', sha256='f819cbfdfb0a7b19c9c52c6f2ca63df0e58a6125f4f139707b586b9511d7fe95', expand=False)
     version('1.12.5', sha256='2b20905942acc580697c8c496d0d1672ab617facb9d30d156b3c7676e67902ec', expand=False)
 
-    depends_on('ruby@2.5.0:', type=('build', 'run'))
-    depends_on('ruby-racc@1.8.1:', type=('build','run'))
-    depends_on('ruby-mini-portile2@2.6.1', type=('build', 'run'))
+    with when('@1.12.5'):
+       depends_on('ruby@2.5.0:3.0.0', type=('build', 'run'))
+       depends_on('ruby-racc@1.4.0:', type=('build','run'))
+       depends_on('ruby-mini-portile2@2.6.1:2.7.0', type=('build', 'run'))
+
+    with when('@1.16.7'):
+       depends_on('ruby@3.0.0:', type=('build', 'run'))
+       depends_on('ruby-racc@1.4.0:', type=('build','run'))
```

Comment:
This one seems to be the same for both

Reply:
Yeah, I just copied and pasted the `https://rubygems.org/gems/nokogiri/versions/1.16.7/dependencies`, but we can refactor it.

Reply:
Sorry here: https://rubygems.org/gems/nokogiri/